services/src/test/scala/cromwell/pipeline/service/GitLabProjectVersioningTest.scala

```diff
@@ -31,20 +41,74 @@ class GitLabProjectVersioningTest extends AsyncWordSpec with ScalaFutures with M
           _ shouldBe VersioningException("Could not create a repository for deleted project.")
         }
       }
+
       "return new active Project with 201 response" taggedAs Service in {
         val project = projectWithRepo
         when(request(activeProject)).thenReturn(Future.successful(Response(StatusCodes.Created.intValue, "", Map())))
         gitLabProjectVersioning.createRepository(activeProject).map {
           _ shouldBe Right(project)
         }
       }
+
       "throw new VersioningException with not 400 response" taggedAs Service in {
         when(request(activeProject)).thenReturn(Future.successful(Response(StatusCodes.BadRequest.intValue, "", Map())))
         gitLabProjectVersioning.createRepository(activeProject).map {
           _ shouldBe Left(VersioningException("The repository was not created. Response status: 400"))
         }
       }
     }
+
+    "updateFile" should {
+      "return create file response when file is new" taggedAs Service in {
+        val path = URLEncoder.encode(newFile.path.toString, "UTF-8")
+        val url = s"${gitLabConfig.url}projects/${projectWithRepo.repository}/repository/files/$path"
+
+        when(
+          mockHttpClient.put(
+            url,
+            payload = s"""{"branch":"${gitLabConfig.defaultFileVersion}","content":"${newFile.content}"}""",
```

Comment:
It's better to use here objects instead of strings. 
Json builded with strings should be used in controllers tests.

Reply:
Fixed